### PR TITLE
[AUD-1459] Refactor card lists to FlatList

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Defines `CardList` using `FlatList`, creates `CollectionList` and `ArtistList` using `CardList`, and shares those components in the profile and explore screens.

### CODE REVIEW

File: src/components/collection-list/CollectionList.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { UserCollection } from 'audius-client/src/common/models/Collection'
+import { Text } from 'react-native'
+
+import { CardList, CardListProps } from 'app/components/core'
+import { CollectionCard } from 'app/screens/explore-screen/components/CollectionCard'
+
+import { EmptyTab } from '../../screens/profile-screen/EmptyTab'
```
Comment (sliptype) [inline src/components/collection-list/CollectionList.tsx:7]: Should we move/rename the EmptyTab component?

File: src/components/core/CardList.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+type CardListRenderItemInfo<ItemT> = ListRenderItemInfo<ItemT> & {
+  style: StyleProp<ViewStyle>
+}
+
+export type CardListProps<ItemT> = ListProps<ItemT> & {
+  renderItem: (info: CardListRenderItemInfo<ItemT>) => ReactElement | null
+}
+
+export function CardList<ItemT>(props: CardListProps<ItemT>) {
```
Comment (sliptype) [inline src/components/core/CardList.tsx:22]: Did you make this a named function to allow for the type parameter? Another option (not necessarily better) I've discovered is:

```suggestion
export const CardList = <ItemT, _>(props: CardListProps<ItemT>) => {
```

The unused type param lets JSX figure out that it's not a component

File: src/screens/profile-screen/AlbumsTab.tsx [hunk 3]
```diff
@@ -1,6 +1,9 @@
+import { useMemo } from 'react'
+
 import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
 
-import { CollectionList } from './CollectionList'
+import { CollectionList } from '../../components/collection-list/CollectionList'
```
Comment (sliptype) [inline src/screens/profile-screen/AlbumsTab.tsx:5]:
```suggestion
import { CollectionList } from 'app/components/collection-list/CollectionList'
```

File: src/screens/profile-screen/AlbumsTab.tsx [hunk 4]
```diff
@@ -9,12 +12,17 @@ const messages = {
 
 export const AlbumsTab = () => {
   const { profile, albums } = useSelectorWeb(getProfile)
-  if (!profile || !albums) return null
+
+  const userAlbums = useMemo(() => {
+    if (profile && albums)
```
Comment (sliptype) [inline src/screens/profile-screen/AlbumsTab.tsx:17]: I prefer curlies for single line conditionals, but I don't think we have that in the styleguide

File: src/styles/typography.ts [hunk 5]
```diff
@@ -41,5 +46,10 @@ export const typography = {
     fontFamily: fontByWeight.medium,
     marginBottom: 14 * 0.35
   },
+  body2: {
```
Comment (sliptype) [inline src/styles/typography.ts:49]: `body2` seems kinda ambiguous to me. I guess it does make sense following h1, h2, etc. pattern though

**Review by sliptype — APPROVED** [review]
Love this consolidation

File: src/components/collection-list/CollectionList.tsx (hunk 1 above)
Comment (dylanjeffers) [reply src/components/collection-list/CollectionList.tsx:7]: yeah def! will rename to EmptyListItem i guess?

File: src/components/core/CardList.tsx (hunk 2 above)
Comment (dylanjeffers) [reply src/components/core/CardList.tsx:22]: totally, its just like... ugh why ts... :/ can update if that's the standard!

File: src/screens/profile-screen/AlbumsTab.tsx (hunk 4 above)
Comment (dylanjeffers) [reply src/screens/profile-screen/AlbumsTab.tsx:17]: yeah agreed, was missed!

File: src/styles/typography.ts (hunk 5 above)
Comment (dylanjeffers) [reply src/styles/typography.ts:49]: yeah its... not the best, im kinda throwing out values here to try and categorize stuff, just so we can take stock of which values we are centering on, and then we can bring to julian to nail down naming convention?

File: src/components/core/CardList.tsx (hunk 2 above)
Comment (sliptype) [reply src/components/core/CardList.tsx:22]: It's not a standard! both ways are annoying for sure

**Pull request merged**